### ISSUE TITLE
Unsorted data crash in gamma codec

### ISSUE THREAD

**Issue by jkbonfield** [enasequence/cramtools#31]
```
jkb@seq3a[htslib/test] cat a.sam
@SQ     SN:xx   LN:20
s2      0       xx      11      1       10M     *       0       0       TTTTTTTTTT      *
s3      0       xx      1       1       10M     *       0       0       AAAAAAAAAA      *
jkb@seq3a[htslib/test] java -Xmx4000m -jar ~/work/cram/cramtools/cramtools-3.0.jar cram -R xx.fa -I a.sam -O _.cram -l debug
DEBUG   2015-04-09 17:18:38     Bam2Cram        Creating tracks for reference: name=xx, length=20.

INFO    2015-04-09 17:18:38     FixBAMFileHeader        Reference sequence MD5 not found, calculating: xx
DEBUG   2015-04-09 17:18:38     Bam2Cram        Writing 2 records for sequence 0, xx
INFO    2015-04-09 17:18:38     Bam2Cram        create: tracks 0ms, records 1ms, mating 0ms.
DEBUG   2015-04-09 17:18:38     CompressionHeaderFactory        Assigned external id to bases: 0
DEBUG   2015-04-09 17:18:38     CompressionHeaderFactory        Assigned external id to quality scores: 1
DEBUG   2015-04-09 17:18:38     CompressionHeaderFactory        Assigned external id to read names: 2
DEBUG   2015-04-09 17:18:38     CompressionHeaderFactory        Assigned external id to mate info: 3
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at net.sf.cram.CramTools.invoke(CramTools.java:93)
        at net.sf.cram.CramTools.main(CramTools.java:123)
Caused by: java.lang.RuntimeException: Invalid valid: -9
        at net.sf.cram.encoding.GammaIntegerCodec.numberOfBits(GammaIntegerCodec.java:68)
        at net.sf.cram.encoding.GammaIntegerCodec.numberOfBits(GammaIntegerCodec.java:23)
        at net.sf.cram.build.CompressionHeaderFactory$EncodingLengthCalculator.add(CompressionHeaderFactory.java:750)
        at net.sf.cram.build.CompressionHeaderFactory$IntegerEncodingCalculator.addValue(CompressionHeaderFactory.java:810)
        at net.sf.cram.build.CompressionHeaderFactory.getOptmialIntegerEncoding(CompressionHeaderFactory.java:618)
        at net.sf.cram.build.CompressionHeaderFactory.build(CompressionHeaderFactory.java:129)
        at net.sf.cram.build.ContainerFactory.buildContainer(ContainerFactory.java:67)
        at net.sf.cram.build.ContainerFactory.buildContainer(ContainerFactory.java:58)
        at net.sf.cram.Bam2Cram.main(Bam2Cram.java:369)
        ... 6 more
```

If I swap the two reads around to make them sorted, so pos 1, pos 11, then it works.

I am unsure if this is just lack of support for unsorted data or if it is just triggering a bug somewhere else.

Cramtools can decode this file if it is created by htslib.

**Comment by vadimzalunin** [comment]
due to lack of support of unsorted reads: gamma codec expects positive offsets. 
fix: b2c883bb2ad00a6988215a4fe283e2aa18d2a878

**Comment by jkbonfield** [comment]
Verified it's working.
